channels/drive/client/drive_file.c

```diff
@@ -137,95 +138,30 @@ static WCHAR* drive_file_combine_fullpath(const WCHAR* base_path, const WCHAR* p
 	return fullpath;
 }
 
-static BOOL drive_file_remove_dir(const WCHAR* path)
-{
-	WIN32_FIND_DATAW findFileData = { 0 };
-	BOOL ret = TRUE;
-	HANDLE dir = INVALID_HANDLE_VALUE;
-	WCHAR* fullpath = NULL;
-	WCHAR* path_slash = NULL;
-	size_t base_path_length = 0;
-
-	if (!path)
-		return FALSE;
-
-	base_path_length = _wcslen(path);
-	path_slash = (WCHAR*)calloc(base_path_length + 3, sizeof(WCHAR));
-
-	if (!path_slash)
-	{
-		WLog_ERR(TAG, "malloc failed!");
-		return FALSE;
-	}
-
-	CopyMemory(path_slash, path, base_path_length * sizeof(WCHAR));
-	path_slash[base_path_length] = L'/';
-	path_slash[base_path_length + 1] = L'*';
-	DEBUG_WSTR("Search in %s", path_slash);
-	dir = FindFirstFileW(path_slash, &findFileData);
-
-	if (dir == INVALID_HANDLE_VALUE)
-	{
-		free(path_slash);
-		return FALSE;
-	}
-
-	do
-	{
-		const size_t len = _wcsnlen(findFileData.cFileName, ARRAYSIZE(findFileData.cFileName));
-
-		if ((len == 1 && findFileData.cFileName[0] == L'.') ||
-		    (len == 2 && findFileData.cFileName[0] == L'.' && findFileData.cFileName[1] == L'.'))
-		{
-			continue;
-		}
-
-		fullpath = drive_file_combine_fullpath(path_slash, findFileData.cFileName, len);
-		DEBUG_WSTR("Delete %s", fullpath);
-
-		if (findFileData.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY)
-		{
-			ret = drive_file_remove_dir(fullpath);
-		}
-		else
-		{
-			ret = DeleteFileW(fullpath);
-		}
-
-		free(fullpath);
-
-		if (!ret)
-			break;
-	} while (ret && FindNextFileW(dir, &findFileData) != 0);
-
-	FindClose(dir);
-
-	if (ret)
-	{
-		if (!RemoveDirectoryW(path))
-		{
-			ret = FALSE;
-		}
-	}
-
-	free(path_slash);
-	return ret;
-}
-
 static BOOL drive_file_set_fullpath(DRIVE_FILE* file, WCHAR* fullpath)
 {
 	if (!file || !fullpath)
 		return FALSE;
 
+	const size_t len = _wcslen(fullpath);
 	free(file->fullpath);
+	free(file->filename);
+	file->fullpath = NULL;
+	file->filename = NULL;
+	if (len == 0)
+		return TRUE;
+
 	file->fullpath = fullpath;
-	file->filename = _wcsrchr(file->fullpath, L'/');
 
-	if (file->filename == NULL)
-		file->filename = file->fullpath;
-	else
-		file->filename += 1;
+	const WCHAR sep[] = { PathGetSeparatorW(PATH_STYLE_NATIVE), '\0' };
+	WCHAR* filename = _wcsrchr(file->fullpath, *sep);
+	if (_wcsncmp(filename, sep, ARRAYSIZE(sep)) == 0)
+		*filename = '\0';
 
+	if (!filename || (_wcsnlen(filename, 2) <= 1))
+		file->filename = _wcsdup(file->fullpath);
```

Comment:
Since we are touching this code - filename is rather useless (it's just allocated and freed)

Reply:
good point, removed.